iquip/apps/dataviewer.py

```diff
@@ -156,3 +161,80 @@ def setData(self, data: np.ndarray, axes: Sequence[AxisInfo]):
         x, y = haxis.values[0], vaxis.values[0]
         width, height = haxis.values[-1] - x, vaxis.values[-1] - y
         self.image.setRect(x, y, width, height)
+
+
+class _RealtimePart(QWidget):
+    """Part widget for configuring realtime mode of the source widget.
+    
+    Attributes:
+        label: Label for showing information about the current experiment.
+          When it is synchronized with an experiment, it displays the RID
+          of the experiment. Otherwise, it shows "No running experiment.".
+        button: Button for synchronizing with the current artiq master.
+    """
+
+    def __init__(self, parent=None):
+        """Extended."""
+        super().__init__(parent=parent)
+        self.label = QLabel("No running experiment.", self)
+        self.button = QPushButton("Sync", self)
+        layout = QHBoxLayout(self)
+        layout.addWidget(self.label)
+        layout.addWidget(self.button)
+
+
+class _RemotePart(QWidget):
+    """Part widget for configuring remote mode of the source widget.
+    
+    Attributes:
+        spinbox: Spinbox for RID input.
+        label: Label for showing the execution time of the experiment.
+    """
+
+    def __init__(self, parent=None):
+        """Extended."""
+        super().__init__(parent=parent)
+        self.spinbox = QSpinBox(self)
+        self.spinbox.setRange(0, 999_999_999)
+        self.spinbox.setButtonSymbols(QAbstractSpinBox.NoButtons)
```

Comment:
Why did you set this to `NoButtons`? Is it a temporary operation?

Reply:
I don't think it requires +/- buttons. Note that we can use up/down keys to increase/decrease the current value of the spinbox.
Also, it is a temporary solution as well. Later I want to synchronize it to the result explorer app to select the RID from its list.

Reply:
> Note that we can use up/down keys to increase/decrease the current value of the spinbox.

I didn't think this way!

Reply:
> Also, it is a temporary solution as well. Later I want to synchronize it to the result explorer app to select the RID from its list.

👍